core/src/main/java/org/apache/iceberg/io/ClusteredPositionDeleteWriter.java

```diff
@@ -46,17 +49,39 @@ public ClusteredPositionDeleteWriter(
       OutputFileFactory fileFactory,
       FileIO io,
       long targetFileSizeInBytes) {
+    this(writerFactory, fileFactory, io, targetFileSizeInBytes, DeleteGranularity.PARTITION);
+  }
+
+  public ClusteredPositionDeleteWriter(
+      FileWriterFactory<T> writerFactory,
+      OutputFileFactory fileFactory,
+      FileIO io,
+      long targetFileSizeInBytes,
+      DeleteGranularity granularity) {
     this.writerFactory = writerFactory;
     this.fileFactory = fileFactory;
     this.io = io;
     this.targetFileSizeInBytes = targetFileSizeInBytes;
+    this.granularity = granularity;
     this.deleteFiles = Lists.newArrayList();
     this.referencedDataFiles = CharSequenceSet.empty();
   }
 
   @Override
   protected FileWriter<PositionDelete<T>, DeleteWriteResult> newWriter(
       PartitionSpec spec, StructLike partition) {
+    switch (granularity) {
+      case FILE:
+        return new TargetedPositionDeleteWriter<>(() -> newRollingWriter(spec, partition));
```

Comment:
This doesn't need to use rolling writers, right?

Reply:
We can actually roll correctly here because this is the "clustered" path. We are not going to use the sorting writer and will not buffer deletes.